### ISSUE TITLE
softmax_cross_entropy_with_logits_v2 -> sparse_softmax_cross_entropy_with_logits

### ISSUE THREAD

**Issue by ilyamironov** [tensorflow/privacy#11]
Replace softmax_cross_entropy_with_logits_v2 in cnn_model_fn (mnist_vizier.py) with sparse_softmax_cross_entropy_with_logits.

It does not change the numerical value of the loss function, but it simplifies the code (no need to transform labels into one-hot vectors) and it is conceptually a better fit for the task at hand (since the classes are disjoint).
